crates/bevy_pbr/src/material.rs

```diff
@@ -245,11 +245,9 @@ impl<M: SpecializedMaterial> SpecializedMeshPipeline for MaterialPipeline<M> {
         if let Some(fragment_shader) = &self.fragment_shader {
             descriptor.fragment.as_mut().unwrap().shader = fragment_shader.clone();
         }
-        descriptor.layout = Some(vec![
-            self.mesh_pipeline.view_layout.clone(),
-            self.material_layout.clone(),
-            self.mesh_pipeline.mesh_layout.clone(),
-        ]);
+
+        let descriptor_layout = descriptor.layout.as_mut().unwrap();
```

Comment:
Why is this `unwrap` here? Is it safe? Not clear how we went from infallible to fallible here...

Reply:
`MeshPipeline::specialize`'s current implementation guarentees that the returned specialized descriptor has a populated `layout`.

Reply:
Thanks! That's what I was looking for. How about:

```suggestion
        let descriptor_layout = descriptor.layout.as_mut().expect("MeshPipeline::specialize's current implementation guarantees that the returned specialized descriptor has a populated layout.");
```

Reply:
Not a fan of using expect and adding more strings to a binary when we know for sure it's going to be valid. expect is a useful tool for panicking when a user makes a faulty error, but this is entirely internal.

Reply:
It makes sense; how about just leaving a comment then?